feat: add `blas/base/wasm/dznrm2`

ShabiShett07 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

none

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds the package `blas/base/wasm/dznrm2`
- replicates the merged package `blas/base/wasm/scnrm2`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add blas/base/wasm/dznrm2

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/dznrm2/test/test.main.js:
```diff
@@ -0,0 +1,170 @@
+* @param {Collection} expected - expected values
+* @param {number} rtol - relative tolerance
+*/
+function isApprox( t, actual, expected, rtol ) {
```
All the same comments as for the PR for `scnrm2`.

ShabiShett07 committed:
chore: fix test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, made necessary changes as per mentioned in the `scnrm2` PR

ShabiShett07 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/dznrm2

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/dznrm2/README.md:
````diff
@@ -0,0 +1,315 @@
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var oneTo = require( '@stdlib/array/one-to' );
+var Complex128Array = require( '@stdlib/array/complex128' );
+var dznrm2 = require( '@stdlib/blas/base/wasm/dznrm2' );
+
+function main() {
+    var xbuf;
+    var out;
+    var x;
+    var N;
+
+    // Specify a vector length:
+    N = 5;
+
+    // Create an input array:
+    xbuf = oneTo( N*2, 'float64' );
+    x = new Complex128Array( xbuf.buffer );
+
+    // Perform computation:
+    out = dznrm2.ndarray( N, x, 1, 0 );
+
+    // Print the result:
+    console.log( out );
+}
````
```suggestion
var oneTo = require( '@stdlib/array/one-to' );
var Complex128Array = require( '@stdlib/array/complex128' );
var dznrm2 = require( '@stdlib/blas/base/wasm/dznrm2' );

// Specify a vector length:
var N = 5;

// Create an input array:
var xbuf = oneTo( N*2, 'float64' );
var x = new Complex128Array( xbuf.buffer );

// Perform computation:
var out = dznrm2.ndarray( N, x, 1, 0 );

// Print the result:
console.log( out );
```

kgryte committed:
docs: fix example

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/dznrm2/package.json:
```diff
@@ -0,0 +1,79 @@
+{
+  "name": "@stdlib/blas/base/wasm/dznrm2",
+  "version": "0.0.0",
+  "description": "Multiply a vector `x` by a scalar `alpha`.",
```
```suggestion
  "description": "Compute the L2-norm of a complex double-precision floating-point vector.",
```

kgryte committed:
docs: fix description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/dznrm2/src/main.wat:
```diff
@@ -0,0 +1,261 @@
+(module
```
```suggestion
;; @license Apache-2.0
;;
;; Copyright (c) 2025 The Stdlib Authors.
;;
;; Licensed under the Apache License, Version 2.0 (the "License");
;; you may not use this file except in compliance with the License.
;; You may obtain a copy of the License at
;;
;;    http://www.apache.org/licenses/LICENSE-2.0
;;
;; Unless required by applicable law or agreed to in writing, software
;; distributed under the License is distributed on an "AS IS" BASIS,
;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
;; See the License for the specific language governing permissions and
;; limitations under the License.

(module
```

kgryte committed:
docs: add missing license

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
